modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -205,14 +208,28 @@ public MetadataMediator(RuntimeExtensionModel extensionModel, RuntimeComponentMo
      */
     private TypeMetadataDescriptor buildParameterTypeMetadataDescriptor(ParameterModel parameterModel)
     {
-        List<MetadataType> subTypes = subTypesMappingContainer.getSubTypes(parameterModel.getType());
+        MetadataType parameterType = parameterModel.getType();
+        TypeMetadataDescriptorBuilder typeDescriptorBuilder = typeDescriptor(parameterModel.getName());
+
+        List<MetadataType> subTypes = subTypesMappingContainer.getSubTypes(parameterType);
         if (!subTypes.isEmpty())
```

Comment:
revert this... if it's empty then return early instead of returning at the end of the if block